presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/TarballController.java

```
@@ -0,0 +1,138 @@
+            }
+        }
+        LOGGER.debug("Successfully installed Presto");
+    }
+
+    private void tarInstall(File tarFile)
+            throws PrestoManagerException
+    {
+        try {
+            if (isDirectory(TAR_INSTALLATION_DIR)) {
+                throw new PrestoManagerException(format("Directory '%s' already exist", TAR_INSTALLATION_DIR.toString()));
+            }
+            else {
+                createDirectories(TAR_INSTALLATION_DIR);
+            }
+            untarFile(tarFile, TAR_INSTALLATION_DIR);
+            createDirectories(CONFIG_DIR);
+        }
+        catch (IOException e) {
+            throw new PrestoManagerException("Failed to install Presto");
+        }
+    }
+
+    public void uninstallAsync(boolean checkDependencies)
+            throws PrestoManagerException
+    {
+        try {
+            deleteRecursively(TAR_INSTALLATION_DIR);
+            deleteRecursively(TAR_DATA_DIR);
+        }
+        catch (IOException e) {
+            throw new PrestoManagerException("Failed to uninstall Presto", e);
+        }
+        LOGGER.debug("Successfully uninstalled Presto");
+    }
+
+    public void upgradePackage(File tarFile, boolean checkDependencies)
+            throws PrestoManagerException
+    {
+        uninstallAsync(checkDependencies);
+        tarInstall(tarFile);
+    }
+
+    public File getPackageFile(URL packageUrl)
+            throws PrestoManagerException
+    {
+        File tarTempFile;
+        try {
+            LOGGER.debug("Downloading file from url: %s", packageUrl);
+            tarTempFile = downloadFile(packageUrl).toFile();
+        }
+        catch (IOException e) {
+            throw new PrestoManagerException(format("Failed to download the file from url : %s", packageUrl.toString()));
+        }
+        return tarTempFile;
+    }
+
+    public void startAsync()
+            throws PrestoManagerException
+    {
+        throw new PrestoManagerException("Unsupported operation");
```

If these are temporary placeholders, they should throw `UnsupportedOperationException("Not implemented")`, and have TODOs to implement.